libc/mem/unveil.c

```diff
@@ -135,7 +132,49 @@ static int sys_unveil_linux(const char *path, const char *permissions) {
 }
 
 /**
- * Unveil parts of a restricted filesystem view.
+ * Restricts filesystem operations, e.g.
+ *
+ *    unveil("/etc", "r");
+ *
+ * Unveiling restricts the visibility of the filesystem to a set of allowed
+ * paths with specific operations. This system call is supported natively on
+ * OpenBSD and polyfilled on Linux using the Landlock LSM[1].
+ *
+ * On OpenBSD, accessing paths outside of the allowed set raises ENOENT, and
+ * accessing ones with incorrect permissions raises EACCES. On Linux, both these
+ * cases raise EACCES.
+ *
+ * Using unveil is irreversible. On OpenBSD, the first call immediately enforces
+ * the filesystem visibilty, and existing paths can only be updated with equal
+ * or lesser permissions. Filesystem operations that try to access invisible
+ * paths will raise ENOENT, and operations without the correct permissions raise
+ * EACCES. Unveiling can be disabled by either passing two NULL arguments or by
+ * calling pledge() without the "unveil" promise.
+ *
+ * Landlock is more permissive than OpenBSD's unveil. Filesystem visibility is
+ * only enforced after disabling, and path permissions can be increased at any
```

Comment:
What does this mean path permissions can be increased at any time? Does that mean it's reversible?

Reply:
I'll reword this later, but it's what your tests show policy union: The most permissive rule takes for a path is used. This means that this idiom doesn't work like on OpenBSD:

```c
unveil("/path", "rwc");
unveil("/path", "r");
// ...
open("/path", O_WRONLY|O_CREAT); // wut
```